### ISSUE TITLE
[Afreecatv] Unable to directly download Subsciption-Only videos

### ISSUE THREAD

**Issue by SingleRottenChips** [yt-dlp/yt-dlp#8580]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.10.13** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Canada & China

### Provide a description that is worded well enough to be understood

I cannot download the Subsciption-Only videos from AfreecaTV with the _video url_.

It stops at "Downloading video info XML"  and reports an error:

> ERROR: [afreecatv] 104647403: afreecatv said: SUBSCRIBE

I _have_ subcribed to the streamer, and I can freely watch those videos on Chrome.

However, yt-dlp _can_ extract the "master" m3u8 file - which I got from browser extension - and list formats I can choose to download.  Yet it _cannot_ deal with the m3u8 url of the video which is playing, and this type of urls can usually be extracted by m3u8 dowloaders.

The video url is like: 
https://vod.afreecatv.com/player/104647403

The "master" m3u8 url: 
https://vod-normal-global-cdn-z02.afreecatv.com/v101/afflv14/_definst_/smil:save/afreeca/station/2023/0714/00/xxxxxxxxxxxxxxx.smil/playlist.m3u8

And the other m3u8:
https://vod-normal-global-cdn-z02.afreecatv.com/v101/hls/save/afreeca/station/2023/0714/00/xxxxxxxxxxxxxxx.smil/original/both/playlist.m3u8

It looks confusing...

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU -F https://vod.afreecatv.com/player/104647403
[debug] Command-line config: ['-vU', '-F', 'https://vod.afreecatv.com/player/104647403']
[debug] Home config "yt-dlp.conf": ['-f', 'bv*[protocol!*=dash]+ba/b', '--write-thumbnail', '--convert-thumbnails', 'jpg', '--write-sub', '--sub-langs', 'en,zh-Hans,ja', '--hls-prefer-native', '--username', 'PRIVATE', '--password', 'PRIVATE']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.10.13 [b634ba742] (pip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.4.0-166-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, certifi-2019.11.28, mutagen-1.47.0, secretstorage-2.3.1, sqlite3-3.31.1, websockets-12.0
[debug] Proxy map: {}
[debug] Loaded 1890 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.10.13, Current version: stable@2023.10.13
yt-dlp is up to date (stable@2023.10.13)
[afreecatv] Logging in
[afreecatv] Extracting URL: https://vod.afreecatv.com/player/104647403
[afreecatv] 104647403: Downloading JSON metadata
[afreecatv] 104647403: Downloading video info XML
ERROR: [afreecatv] 104647403: afreecatv said: SUBSCRIBE
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/extractor/common.py", line 715, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/extractor/afreecatv.py", line 227, in _real_extract
    raise ExtractorError(
```

**Comment by bashonly** [comment]
You need to [pass cookies to yt-dlp](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp)

Try that, and if it still doesn't work, please share a new verbose log.

Also, the m3u8 URLs you shared are not useful information if you redact segments of them and break them

**Comment by SingleRottenChips** [comment]
> You need to [pass cookies to yt-dlp](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp)
> 
> Note that per [this comment](https://github.com/yt-dlp/yt-dlp/issues/4514#issuecomment-1313448134), `--cookies-from-browser` may not work with this site, in which case you will need to manually export your cookies to a cookiefile to be used with the `--cookies` option.
> 
> Try that, and if it still doesn't work, please share a new verbose log (w/ passing cookies).
> 
> Also, the m3u8 URLs you shared are not useful information if you redact segments of them and break them

Thanks for your reminder. 
I just tried to remove --cookies cookies.txt, but now I add this line back and update my cookies, the result is the same.

```
 yt-dlp -vU -F https://vod.afreecatv.com/player/104647403
[debug] Command-line config: ['-vU', '-F', 'https://vod.afreecatv.com/player/104647403']
[debug] Home config "yt-dlp.conf": ['-f', 'bv*[protocol!*=dash]+ba/b', '--write-thumbnail', '--convert-thumbnails', 'jpg', '--cookies', 'cookies.txt', '--write-sub', '--sub-langs', 'en,zh-Hans,ja', '--hls-prefer-native', '--username', 'PRIVATE', '--password', 'PRIVATE']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.10.13 [b634ba742] (pip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.4.0-166-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, certifi-2019.11.28, mutagen-1.47.0, secretstorage-2.3.1, sqlite3-3.31.1, websockets-12.0
[debug] Proxy map: {}
[debug] Loaded 1890 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.10.13, Current version: stable@2023.10.13
yt-dlp is up to date (stable@2023.10.13)
[afreecatv] Logging in
[afreecatv] Extracting URL: https://vod.afreecatv.com/player/104647403
[afreecatv] 104647403: Downloading JSON metadata
[afreecatv] 104647403: Downloading video info XML
ERROR: [afreecatv] 104647403: afreecatv said: SUBSCRIBE
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/extractor/common.py", line 715, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/extractor/afreecatv.py", line 227, in _real_extract
    raise ExtractorError(
```

I updated my cookies by copying it from the extension, since I'm using a server and I don't know which browser to choose (Could you tell me how to write it? Thank you). 

I'm wondering whether the cookies are necessary here since I _have_ passed my username and password (You can see "[afreecatv] Logging in" in the debug). And actually, I _am_ able to download 19+ videos with and without cookies on my latop and the server with username and password

As for m3u8 urls, yeah, I changed the number in original m3u8 url to xxxxxxxx. And I just realized that yt-dlp is able to handle the master url (which has no video quality specified), while m3u8 downloaders (software, website, extension...) deal with the url of a specific format ("original", "hd4k"...).

Is it possible that afreecatv does not give us the permission to extract the info of Subsciption-Only videos?
